server/src/main/java/org/apache/seata/server/session/GlobalSession.java

```diff
@@ -203,13 +211,33 @@ public boolean isTimeout() {
     }
 
     /**
-     * prevent could not handle committing and rollbacking transaction
+     * prevent could not handle committing, rollbacking and rollbacked transaction
+     *
+     * If the global session's status is in end status, it returns the remaining time until the session reaches
+     * the end state retry dead threshold.
+     * For other statuses, it returns the remaining time until the session reaches the retry dead threshold.
+     *
      * @return time to dead session. if not greater than 0, then deadSession
      */
     public long timeToDeadSession() {
+        if (isEndStatus()) {
+            return beginTime + END_STATE_RETRY_DEAD_THRESHOLD - System.currentTimeMillis();
+        }
         return beginTime + RETRY_DEAD_THRESHOLD - System.currentTimeMillis();
     }
 
+    private boolean isEndStatus() {
+        EnumSet<GlobalStatus> deadStatuses = EnumSet.of(
+            GlobalStatus.Rollbacked,
+            GlobalStatus.TimeoutRollbacked,
+            GlobalStatus.CommitFailed,
+            GlobalStatus.RollbackFailed,
+            GlobalStatus.TimeoutRollbackFailed,
+            GlobalStatus.Finished
+        );
```

Comment:
```suggestion
        EnumSet<GlobalStatus> endStatuses = EnumSet.of(
            GlobalStatus.Rollbacked,
            GlobalStatus.Committed,
            GlobalStatus.Finished
        );
```

@funky-eyes 

WDYT?

Reply:
This is okay

Reply:
Sorry, I think I missed TimeoutRollbacked. Since TimeoutRollbacked is not a failed end state, it should be handled by scheduled tasks.

Reply:
No worries 😉
I'll fix it.